server/controllers/finance/debtorGroups.js

```diff
@@ -111,20 +111,26 @@ function getDetail(req, res, next) {
 
 /**
 * GET /debtor_groups/ => all debtor groups
-* GET /debtor_groups/?locked={1|0} => only locked or not locked debtor groups
+* GET /debtor_groups/?locked={1|0}&is_convention={1|0} => locked or convention criteria
 *
 * @function getList
 *
 * @desc This function is responsible for retrieving details of a debtor group
 */
 function getList(req, res, next) {
-  var query;
+  var query, criteria, conditions = [];
 
-  query = 'SELECT uuid, name, locked, account_id FROM debitor_group ';
-  query += (req.query.locked === '1') ? 'WHERE locked = 1' :
-           (req.query.locked === '0') ? 'WHERE locked = 0' : '';
+  query = 'SELECT uuid, name, locked, account_id, is_convention FROM debitor_group ';
 
-  db.exec(query)
+  /** build the where clause criteria */
+  criteria = Object.keys(req.query).map(function (item) {
```

Comment:
I think you could write this code differently to let MySQL take care of escaping all columns.  Try this:
```js
criteria = Object.keys(req.query).map(function (item) {
  conditions = conditions.concat(item, req.query[item]));
  return '?? = ?';
}). join (' AND ');
```
This will produce an SQL string and conditions  like this:
```js
var sql = 'SELECT column FROM table WHERE ?? = ? AND ?? = ?';
var conditions = ['id', 1, 'name', 'jack'];

// using db.exec() to template this string will result in
// SELECT column FROM table WHERE `id` = 1 AND `name` = 'jack';
```
That way, MySQL does all the templating and query escaping.   If this works, it would be best to move it into its own function to process query string conditions.